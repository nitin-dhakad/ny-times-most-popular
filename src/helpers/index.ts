/**
 * write a method to get publish day and month
 * @param {*} url function take as input url
 * @param {*} params function n params in order to replace in API
 * @return {*} updated url with provied parameters
 */

export const updateURLParams = (url: string, ...params: string[]) => {
  let updatedUrl = url;
  params.forEach((param, index) => {
    updatedUrl = updatedUrl.replace(`{{param${index + 1}}}`, param);
  });
  return updatedUrl;
};

/**
 * write a method to fetch media image url
 * @param {*} media function take as input
 * @param {*} quality image quality to be fetched expected values 0 - low , 1- good, 2 - best
 * @return {*} media image url
 */
export const extractImageUrl = (
  media: { [x: string]: { [x: string]: string }[] }[],
  quality: number
): string => {
  return media?.[0]?.["media-metadata"]?.[quality]?.["url"] || "";
};

/**
 * write a method to get publish day and month
 * @param {*} date function take as input published date
 * @return {*} day and month
 */
export const getDateAndMonth = (date: string): string => {
  const dateObj = new Date(date);
  const month = dateObj.toLocaleString("default", { month: "long" });
  const day = dateObj.getDate();
  return day + " " + month;
};
