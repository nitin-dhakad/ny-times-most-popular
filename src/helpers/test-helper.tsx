import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { setupStore } from "../store";



export const renderWithProviders = (
  ui: React.ReactElement,
  { route = "/", ...renderOptions } = {}
) => {
  window.history.pushState({}, "Test page", route);
  return render(
    <Provider store={setupStore()}>
      <BrowserRouter>{ui}</BrowserRouter>
    </Provider>,
    renderOptions
  );
};
