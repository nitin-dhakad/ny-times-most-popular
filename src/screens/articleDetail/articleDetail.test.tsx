import ArticleDetail from ".";
import { renderWithProviders } from "../../helpers/test-helper";

describe("rendering article Detail page", () => {
  test("show article detail page", () => {
    const { container } = renderWithProviders(<ArticleDetail />);
    expect(container).toMatchSnapshot();
  });
});
