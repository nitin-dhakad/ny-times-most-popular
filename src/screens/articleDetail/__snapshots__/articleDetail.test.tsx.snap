// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering article Detail page show article detail page 1`] = `
<div>
  <div>
    <button
      class="absolute top-4 left-6 flex items-center px-3 py-2 text-sm font-semibold text-gray-700 bg-white border border-gray-300 rounded-lg shadow hover:bg-gray-100 focus:outline-none focus:ring focus:ring-gray-200"
    >
      <svg
        class="w-4 h-4 mr-1"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 12H3m0 0l7 7m-7-7l7-7"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      Back
    </button>
    <h1
      class="text-4xl font-semibold font-serif italic text-center p-4 text-slate-800 outline"
    >
      The New York Times
    </h1>
    <div
      class="bg-white  rounded-lg overflow-hidden mx-auto md:flex-row md:flex px-5 pt-2"
    >
      <img
        class="w-full h-full object-cover md:w-6/12 md:max-h-80"
        src="/placeHolder.png"
      />
      <div
        class="p-4"
      >
        <h2
          class="text-xl font-semibold mb-2"
        />
        <p
          class="text-gray-600 text-sm mb-2"
        />
        <p
          class="text-gray-700 mb-4"
        />
        <div
          class="flex justify-between text-gray-500 text-xs"
        >
          <span>
            NaN Invalid Date
          </span>
          <span />
        </div>
      </div>
    </div>
    <div
      class="outline-double mt-2"
    />
  </div>
</div>
`;
